test/integration/apiserver/field_validation_test.go

```diff
@@ -0,0 +1,681 @@
+	defer closeFn()
+
+	bodyBytes, err := os.ReadFile("./testdata/deploy-small-unknown-field.json")
+	if err != nil {
+		t.Fatalf("failed to read file: %v", err)
+	}
+	body := []byte(fmt.Sprintf(string(bodyBytes), `"test-deployment"`))
+
+	var testcases = []struct {
+		name         string
+		opts         metav1.CreateOptions
+		errContains  string
+		warnContains string
+	}{
+		{
+			name:        "post-strict-validation",
+			errContains: "unknown field",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Strict",
+			},
+		},
+		{
+			name:         "post-warn-validation",
+			warnContains: "unknown field",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Warn",
+			},
+		},
+		{
+			name: "post-ignore-validation",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Ignore",
+			},
+		},
+		{
+			name: "post-default-ignore-validation",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			req := client.CoreV1().RESTClient().Post().
+				AbsPath("/apis/apps/v1").
+				Namespace("default").
+				Resource("deployments").
+				VersionedParams(&tc.opts, metav1.ParameterCodec)
+			result := req.Body([]byte(body)).Do(context.TODO())
+			if tc.warnContains != "" {
+				warningMatched := false
+				for _, w := range result.Warnings() {
+					if strings.Contains(w.String(), tc.warnContains) {
+						warningMatched = true
+					}
+				}
+				if !warningMatched {
+					t.Fatalf("expected warning to contain: %s, got warnings: %v", tc.warnContains, result.Warnings())
+				}
+			}
+			resBody, err := result.Raw()
+			if err == nil && tc.errContains != "" {
+				t.Fatalf("unexpected post succeeded")
+			}
+			if err != nil && !strings.Contains(string(resBody), tc.errContains) {
+				t.Fatalf("unexpected response: %v", string(resBody))
+			}
+		})
+	}
+}
+
+// TestFieldValidationPut tests PUT requests containing unknown fields with
+// strict and non-strict field validation.
+func TestFieldValidationPut(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.FieldValidation, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	deployName := `"test-deployment"`
+	postBytes, err := os.ReadFile("./testdata/deploy-small.json")
+	if err != nil {
+		t.Fatalf("failed to read file: %v", err)
+	}
+	postBody := []byte(fmt.Sprintf(string(postBytes), deployName))
+
+	if _, err := client.CoreV1().RESTClient().Post().
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Body(postBody).
+		DoRaw(context.TODO()); err != nil {
+		t.Fatalf("failed to create initial deployment: %v", err)
+	}
+
+	putBytes, err := os.ReadFile("./testdata/deploy-small-unknown-field.json")
+	if err != nil {
+		t.Fatalf("failed to read file: %v", err)
+	}
+	putBody := []byte(fmt.Sprintf(string(putBytes), deployName))
+	var testcases = []struct {
+		name         string
+		opts         metav1.UpdateOptions
+		errContains  string
+		warnContains string
```

Comment:
For this test (and another), I don't know if I enjoy the value of table-driven testing here, as I feel this obfuscates what the test does for very limited code re-use.